rest/config.go

```diff
@@ -96,11 +96,47 @@ type ServerConfig struct {
 
 // Bucket configuration elements - used by db, shadow, index
 type BucketConfig struct {
-	Server   *string `json:"server,omitempty"`   // Couchbase server URL
-	Pool     *string `json:"pool,omitempty"`     // Couchbase pool name, default "default"
-	Bucket   *string `json:"bucket,omitempty"`   // Bucket name
-	Username string  `json:"username,omitempty"` // Username for authenticating to server
-	Password string  `json:"password,omitempty"` // Password for authenticating to server
+	Server     *string `json:"server,omitempty"`      // Couchbase server URL
+	Pool       *string `json:"pool,omitempty"`        // Couchbase pool name, default "default"
+	Bucket     *string `json:"bucket,omitempty"`      // Bucket name
+	Username   string  `json:"username,omitempty"`    // Username for authenticating to server
+	Password   string  `json:"password,omitempty"`    // Password for authenticating to server
+	CertPath   string  `json:"certpath,omitempty"`    // Cert path (public key) for X.509 bucket auth
+	KeyPath    string  `json:"keypath,omitempty"`     // Key path (private key) for X.509 bucket auth
+	CACertPath string  `json:"cacertpath,omitempty"`  // Root CA cert path for X.509 bucket auth
```

Comment:
Is it worth mentioning what happens if `CACertPath` is left out?  Eg, does it default to a self-signed cert?

Reply:
I don't think there's a valid scenario where CertPath is specified by CACertPath isn't, but in case there's a scenario I'm not aware of, I don't think we should be doing anything to explicitly exclude that option.